modules/snooker/src/main/ts/ephox/snooker/selection/TableSelection.ts

```diff
@@ -0,0 +1,34 @@
+import { Element as DomElement, HTMLTableCellElement, HTMLTableRowElement, HTMLTableCaptionElement, Node } from '@ephox/dom-globals';
+import { Arr, Options } from '@ephox/katamari';
+import { TableLookup } from '@ephox/snooker';
+import { Attr, Compare, Element, Elements, SelectorFind } from '@ephox/sugar';
+import { Selections } from './Selections';
+import CellOperations from '../queries/CellOperations';
+
+const getSelectionStartFromSelector = <T extends DomElement>(selector: string) => (start: Element<Node>) =>
+  SelectorFind.closest<T>(start, selector);
+
+const getSelectionStartCaption = getSelectionStartFromSelector<HTMLTableCaptionElement>('caption');
+
+const getSelectionStartCell = getSelectionStartFromSelector<HTMLTableCellElement>('th,td');
+
+const getSelectionStartCellOrCaption = getSelectionStartFromSelector<HTMLTableCellElement | HTMLTableCaptionElement>('th,td,caption');
+
+const getCellsFromSelection = (body: any, start: Element<Node>, selectedSelector: string): Element<HTMLTableCellElement>[] =>
+  getSelectionStartCell(start)
+    .map((cell) => CellOperations.selection(cell, Selections(body, () => start, selectedSelector)))
```

Comment:
Should we really be initialising Selections here, or just get it to pass in the Selections instance initialised in the table plugin? Need to look closer

Reply:
@ltrouton I figure you might know?

Reply:
I'd agree we'd be better passing this in than initializing this. It'd also make more sense coupled with that other suggestion I made (making selections return the cell for single).

Reply:
Talked to Luke and he agreed too. Thought he might have had issues with trying to wire it through or something, so I'll keep an eye out for that when changing this.
- [ ] change to pass in selections